metacat-thrift/src/main/java/com/netflix/metacat/thrift/HiveConvertersImpl.java

```diff
@@ -332,15 +329,19 @@ public PartitionDto hiveToMetacatPartition(final TableDto tableDto, final Partit
             throw new IllegalArgumentException("Invalid partition name", e);
         }
 
-        final List<String> partVals = Lists.newArrayList();
-        for (String key : tableDto.getPartition_keys()) {
-            final String val = hm.get(key);
-            if (val == null) {
-                throw new IllegalArgumentException("Invalid partition name - missing " + key);
+        if (tableDto != null && tableDto.getPartition_keys() != null) {
```

Comment:
`getPartition_keys()` is pretty far from standard java POJO naming convention, is there a reason for this?

Also since you call `getPartition_keys()` here and below why not write it to a local variable first outside the loop that way if the `get` is doing any kind of "expensive" computation you only have to do it once. Also if you do that you can allocate your `partVals` array list with a fixed size

Reply:
Probably the name was unintentionally copied from the legacy TableDto class. Good point on the get call. Will change it.